AEPAnalytics/Sources/Analytics.swift

```diff
@@ -61,57 +66,111 @@ public class Analytics: NSObject, Extension {
         }
 
         Log.trace(label: LOG_TAG, "handleAnalyticsRequest - Processing event with id \(event.id.uuidString).")
-        track(event: event)
+        track(event: event, data: event.data)
+    }
+
+    /// Handler for rules engine response events
+    /// - Parameter event: an event containing consequence for processing
+    private func handleRulesEngineResponse(event: Event) {
+        if event.data == nil {
+            Log.trace(label: LOG_TAG, "Event with id \(event.id.uuidString) contained no data, ignoring.")
+            return
+        }
+
+        Log.trace(label: LOG_TAG, "handleRulesEngineResponse - Processing event with id \(event.id.uuidString).")
+
+        guard let consequence = event.data?[AnalyticsConstants.EventDataKeys.TRIGGERED_CONSEQUENCE] as? [String: Any] else {
+            Log.trace(label: LOG_TAG, "handleRulesEngineResponse - Ignoring as missing consequence data for \(event.id.uuidString).")
+            return
+        }
+
+        guard let consequenceType = consequence[AnalyticsConstants.EventDataKeys.TYPE] as? String, consequenceType == AnalyticsConstants.ConsequenceTypes.TRACK else {
+            Log.trace(label: LOG_TAG, "handleRulesEngineResponse - Ignoring as consequence type is not analytics for \(event.id.uuidString).")
+            return
+        }
+
+        guard let _ = consequence[AnalyticsConstants.EventDataKeys.ID] as? String else {
+            Log.trace(label: LOG_TAG, "handleRulesEngineResponse - Ignoring as consequence id is missing for \(event.id.uuidString).")
+            return
+        }
+
+        let consequenceDetail = consequence[AnalyticsConstants.EventDataKeys.DETAIL] as? [String: Any] ?? [:]
+        track(event: event, data: consequenceDetail)
+    }
+
+    /// Handler for configuration response event
+    /// - Parameter event: an event containing configuration response event
+    private func handleConfigurationResponse(event: Event) {
+        if event.data == nil {
+            Log.trace(label: LOG_TAG, "Event with id \(event.id.uuidString) contained no data, ignoring.")
+            return
+        }
+
+        if let privacyStatusStr = event.data?[AnalyticsConstants.Configuration.GLOBAL_CONFIG_PRIVACY] as? String {
+            let privacyStatus = PrivacyStatus(rawValue: privacyStatusStr) ?? PrivacyStatus.unknown
+            if privacyStatus == .optedOut {
+                // Clear persisted ids
+                clearDataStore()
+            }
+        }
     }
 
     /// Process analytics track request
     /// - Parameter event: an event containing track data for processing
-    private func track(event: Event) {
+    /// - Parameter data: track data for processing
```

Comment:
nit: the comments should follow the `Parameters: ` format if there are multiple parameters